sw/airborne/math/pprz_algebra_float.h

```diff
@@ -721,6 +767,58 @@ static inline void float_mat_vect_mul(float *o, float **a, float *b, int m, int
   }
 }
 
+/** a *= k, where k is a scalar value
+ *
+ * a: [m x n]
+ * k: [1 x 1]
+ */
+static inline void float_mat_scale(float **a, float k, int m, int n)
+{
+  int i, j;
+  for (i = 0; i < m; i++) {
+    for (j = 0; j < n; j++) {
+      if (a[i][j] != 0.0) {
```

Comment:
This inequality is not a good idea, as float doesn't represent 0 well. You usually check if it is larger than some epsilon. https://bitbashing.io/comparing-floats.html

 Also, why do you do this check, the next line of computation is fine if a is 0.